cli/delete.go

```diff
@@ -135,19 +140,10 @@ func deleteRecordCommand(t *core.Timetrace) *cobra.Command {
 
 func askForConfirmation() bool {
 	reader := bufio.NewReader(os.Stdin)
-	for {
-		fmt.Fprint(os.Stderr, "Please confirm (Y/N): ")
-		s, _ := reader.ReadString('\n')
-		s = strings.TrimSuffix(s, "\n")
-		s = strings.ToLower(s)
-		if len(s) > 1 {
-			continue
-		}
-		if strings.Compare(s, "n") == 0 {
-			return false
-		} else if strings.Compare(s, "y") == 0 {
-			break
-		}
-	}
-	return true
+	fmt.Fprint(os.Stderr, "Please confirm [y/N]: ")
```

Comment:
Actually your code looks much cleaner. But I am not sure if @dominikbraun intends to remove the for loop.

I think the for loop is not necessary.  So for me it looks ok :-)

Reply:
You can throw it out if you want to.